distributed/client.py

```diff
@@ -5891,9 +5900,9 @@ def futures_of(o, client=None):
             stack.extend(x.__dask_graph__().values())
 
     if client is not None:
-        bad = {f for f in futures if f.cancelled()}
-        if bad:
-            raise CancelledError(bad)
+        cancelled = {f.key: f._state.exception for f in futures if f.cancelled()}
```

Comment:
This might become very large. Ideally, we'd group these by the reason for cancellation.